draft-ietf-ppm-dap.md

```diff
@@ -1463,10 +1461,9 @@ enum {
   hpke_unknown_config_id(3),
   hpke_decrypt_error(4),
   vdaf_prep_error(5),
-  batch_saturated(6),
-  task_expired(7),
-  invalid_message(8),
-  report_too_early(9),
+  task_expired(6),
+  invalid_message(7),
+  report_too_early(8),
```

Comment:
This is a wire-breaking change. We can avoid it by just deleting `batch_saturated` and not renumbering the remaining variants. However this would leave a gap in the variants defined in our draft, which is a bit odd.

I think we should avoid breaking changes if we can. An alternative would be to leave a note here that if we want a new variant before RFC, then the variant should use `6`.

We could take this one step further by being very explicit about what we want in the end:

```
enum {
  batch_collected(0),
  report_replayed(1),
  report_dropped(2),
  hpke_unknown_config_id(3),
  hpke_decrypt_error(4),
  vdaf_prep_error(5),
  /* [TODO Remove or replace this variant.] */
  todo_remove_me(6),
  task_expired(7),
  invalid_message(8),
  report_too_early(9),
  (256)
} PrepareError;
```

Reply:
I suppose -- every new draft is a breaking change because it will, at least, change the domain-separation strings. I suppose it could be argued that shifting the numbers around could lead to bugs in implementations that implement multiple DAP drafts & use the wrong "version" of `PrepareError`; but we could run into similar issues if/when we reuse the number with a new semantic meaning.

That said, I'm happy enough kicking the can down the road, as long as we clean things up for the final RFC.

Reply:
We don't actually need to update the domain separation tags with each version. In fact, in the latest version of the VDAF draft, we didn't update the version tag because there were no breaking changes.